internal/controller/managedzone_controller.go

```diff
@@ -84,6 +85,15 @@ func (r *ManagedZoneReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			logger.Error(err, "Failed to delete parent Zone NS Record")
 			return ctrl.Result{}, err
 		}
+
+		if recordsExist, err := r.hasOwnedRecords(ctx, managedZone); err != nil {
+			logger.Error(err, "Failed to check owned records")
+			return ctrl.Result{}, err
+		} else if recordsExist {
+			logger.Info("ManagedZone deletion awaiting removal of owned DNS records")
+			return ctrl.Result{Requeue: true}, nil
+		}
```

Comment:
Should this re-queue? Do the owner records not cause this to reconcile on change anyway?